go/fn/object.go

```diff
@@ -595,26 +595,27 @@ func (o *KubeObject) resourceIdentifier() *yaml.ResourceIdentifier {
 	}
 }
 
-// IsGVK compares the given group, version, and kind with KubeObject's apiVersion and Kind.
-func (o *KubeObject) IsGVK(group, version, kind string) bool {
-	var apiVersion string
-	switch {
-	case group == "":
-		apiVersion = version
-	default:
-		apiVersion = group + "/" + version
-	}
-
-	if o.GetAPIVersion() == apiVersion && o.GetKind() == kind {
+func match(GVKStr string, oStr string) bool {
+	if GVKStr == "" {
 		return true
 	}
-	if apiVersion == "" && o.GetKind() == kind {
-		return true
-	}
-	if kind == "" && o.GetAPIVersion() == apiVersion {
-		return true
+	return oStr == GVKStr
+}
+
+// IsGVK compares the given group, version, and kind with KubeObject's apiVersion and Kind.
+func (o *KubeObject) IsGVK(group, version, kind string) bool {
```

Comment:
Some general suggestions about improving the readability:
- `match` is normally read as the strings are strict the same, while here we use it with two special cases 1. empty GVKstring matches all. 2. oStr does not have empty string matches all. If we want to use `match` method, we should rename it to something without the misleading.
- GetKind() does not have code checking whether it's empty string or not. So caller methods like IsGVK cannot assumes it's not empty string. It then does not make sense to say mismatch if GetKind() is empty while "kind" is not.    
- We also have a function `ParseGroupVersion`, suggest calling that function. If that function could not handle "/" well, let's change that function instead.
- Maybe you can consider switching the logic and emphasize more on the `matching` code than the string processing code. like 
```go
oGroup, oVersion := o.ParseGroupVersion(o.GetApiVersion)

if o.GetKind() != "" && kind != "" && o.GetKind() != kind{
     return false
if oGroup != "" && group != "" && oGroup != group {
     return false
if oVersion != "" && version != "" && oVersion != version {
     return false
return true
} 
```

Reply:
Somehow I can not find the function `ParseGroupVersion`

Reply:
> GetKind() does not have code checking whether it's empty string or not. So caller methods like IsGVK cannot assumes it's not empty string. It then does not make sense to say mismatch if GetKind() is empty while "kind" is not.

Quick clarification, does this mean if I have a resource with no `kind`, and I do `o.IsGVK("", "", Deployment)`, it should return `true`?

Reply:
The logic is the same as if group is not given (like ConfigMap), it always matches